quad/src/qag_par.rs

```diff
@@ -266,22 +268,28 @@ impl QagPar {
         while last < self.limit {
             let mut to_process = vec![];
             let mut err_sum = 0.0;
+            let mut old_result = vec![0.0; n];
 
             while to_process.len() < 128 && heap.len() != 0 {
                 let old_interval = heap.pop().unwrap();
                 let ((x, y), old_err) = (old_interval.interval, old_interval.err);
+                if x.abs().max(y.abs())
+                    <= (1.0 + 100.0 * EPMACH) * (((x + y) / 2.0).abs() + 1000.0 * UFLOW)
```

Comment:
Can you add a comment to explain this check?

It seems like you're effectively checking if the numbers are close to each other, because:
$$
max(x, y) <= (1 + 100\varepsilon) * (\frac{x+y}{2} + 1000 u_{flow})
$$
so if *the maximum is smaller than the middle point + something + a relative very small fraction*.

Reply:
Even better: make it a function on its own :)